File: ipc_message_queues/sender.py
import sysv_ipc

# Define the same structure as in C++
MESSAGE_TYPE = 1

# Create a unique key for the queue
key = sysv_ipc.ftok("msgqueue", 65)
msg_queue = sysv_ipc.MessageQueue(key, sysv_ipc.IPC_CREAT)

# Send a message
msg_queue.send(b"Hello from Python!", type=MESSAGE_TYPE)
print("Message sent: Hello from Python!")

File: ipc_message_queues/sender.cpp
// Message Queue for Sender Proces
#include <iostream>
#include <sys/ipc.h>
#include <sys/msg.h>
#include <cstring>

struct Message
{
    long messageType;  // Message type
    char payload[100]; // Message content
};

int main()
{
    key_t key = ftok("msgqueue", 65);          // Generate unique key
    int msgid = msgget(key, 0666 | IPC_CREAT); // Create message queue

    Message message;
    message.messageType = 1;                    // Set message type
    strcpy(message.payload, "Hello from C++!"); // Set payload

    // printf("Write Data: ");
    // fgets(message.payload, MAX, stdin);

    msgsnd(msgid, &message, sizeof(message.payload), 0); // Send message
    std::cout << "Message sent: " << message.payload << std::endl;

    return 0;
}